fineract-provider/src/main/java/org/apache/fineract/infrastructure/dataqueries/service/ReadWriteNonCoreDataServiceImpl.java

```diff
@@ -183,6 +192,116 @@ public DatatableData retrieveDatatable(final String datatable) {
+                rowSet = jdbcTemplate.queryForRowSet(sql, new Object[] { finalValueFilter }, argType);
+            } else if ("date".equalsIgnoreCase(filterColumnType)) {
+                int[] argType = new int[1];
+                argType[0] = Types.DATE;
+                try {
+                    rowSet = jdbcTemplate.queryForRowSet(sql, new Object[] { LocalDate.parse(valueFilter, DATA_TABLE_DATE_FORMAT_MYSQL) },
+                            argType); // NOSONAR
+                } catch (DateTimeParseException e) {
+                    List<ApiParameterError> paramErrors = new ArrayList<>();
+                    paramErrors.add(parameterErrorWithValue("400",
+                            "Unsupported input type for datatable query! Use format: 'yyyy-MM-dd'. Column filter: " + filterColumnType,
+                            "valueFilter", valueFilter));
+                    throw new PlatformApiDataValidationException(paramErrors, e);
+                }
+            } else if ("datetime".equals(filterColumnType)) {
+                int[] argType = new int[1];
+                argType[0] = Types.TIMESTAMP;
+                try {
+                    rowSet = jdbcTemplate.queryForRowSet(sql,
+                            new Object[] { LocalDateTime.parse(valueFilter, DATA_TABLE_DATETIME_FORMAT_MYSQL) }, argType); // NOSONAR
+                } catch (DateTimeParseException e) {
+                    List<ApiParameterError> paramErrors = new ArrayList<>();
+                    paramErrors.add(parameterErrorWithValue("400",
+                            "Unsupported input type for datatable query! Use format: 'yyyy-MM-dd HH:mm:ss'. Column filter: "
+                                    + filterColumnType,
+                            "valueFilter", valueFilter));
+                    throw new PlatformApiDataValidationException(paramErrors, e);
+                }
+            } else {
+                rowSet = jdbcTemplate.queryForRowSet(sql, new Object[] { finalValueFilter }); // NOSONAR
+            }
+        }
+
+        List<JsonObject> results = new ArrayList<>();
+        while (rowSet.next()) {
+            JsonObject json = new JsonObject();
+            for (String rcn : resultColumnNames) {
+                Object rowValue = rowSet.getObject(rcn);
+                if (rowValue != null) {
+                    if (rowValue instanceof Character) {
+                        json.addProperty(rcn, (Character) rowValue);
+                    } else if (rowValue instanceof Number) {
+                        json.addProperty(rcn, new BigDecimal(rowValue.toString()));
+                    } else if (rowValue instanceof Boolean) {
+                        json.addProperty(rcn, (Boolean) rowValue);
+                    } else if (rowValue instanceof LocalDateTime) {
+                        json.addProperty(rcn, DATA_TABLE_DATETIME_FORMAT_MYSQL.format((LocalDateTime) rowValue));
```

Comment:
Very misleading to use the MySQL format here. Let's dupe this member and name it properly like DATA_TABLE_RESULT_DATETIME_FORMAT, or something similar

Reply:
done